plugins/put-a-file-with-ftp/impl/src/main/java/org/pentaho/di/job/entries/ftpput/JobEntryFTPPUT.java

```diff
@@ -642,13 +642,19 @@ public Result execute( Result previousResult, int nr ) {
         }
       }
 
-      Path path = Paths.get( realLocalDirectory );
-      if ( !Files.isReadable( path ) || !Files.isWritable( path ) ) {
-        throw new FTPException( BaseMessages.getString( PKG, "JobFTPPUT.LocalDir.NoPermission" ) );
+      boolean isStandardFileSystem = !realLocalDirectory.contains( "://" ) || realLocalDirectory.equalsIgnoreCase( "file" );
+      if ( isStandardFileSystem ) {
+        if ( realLocalDirectory.charAt( 0 ) == '/' ) {
+          realLocalDirectory = realLocalDirectory.substring( 1 );
+        }
```

Comment:
This seems to be making an absolute path into a relative path. That seems sketchy.